lib/puppet/relationship.rb

```diff
@@ -19,10 +19,10 @@ def self.from_data_hash(data)
 
     args = {}
     if event = data["event"]
-      args[:event] = event
+      args[:event] = :"#{event}"
     end
     if callback = data["callback"]
-      args[:callback] = callback
+      args[:callback] = :"#{callback}"
```

Comment:
Well, `false` and `nil` will never get to this line. See `if` statement above. The `data` is supposed to be a `Hash produced by `to_data_hash` so other objects should really not be here unless someone is doing something that is very out of the ordinary.